src/modules/Sidebar/Sidebar.js

```diff
@@ -67,9 +74,33 @@ class Sidebar extends Component {
     this.stopAnimatingTimer = setTimeout(() => this.setState({ animating: false }), duration)
   }
 
+  componentDidMount() {
+    const { visible, closable } = this.props
+
+    if (visible && closable) {
+      document.addEventListener('click', this.handleDocumentClick)
+    }
+  }
+
   componentWillReceiveProps(nextProps) {
     if (nextProps.visible !== this.props.visible) {
       this.startAnimating()
+      if (nextProps.visible && this.props.closable) {
+        document.addEventListener('click', this.handleDocumentClick)
+      } else {
+        document.removeEventListener('click', this.handleDocumentClick)
+      }
+    }
+  }
+
+  handleRef = c => { this.ref = c }
+
+  handleDocumentClick = (e) => {
+    const sidebarDOMNode = ReactDOM.findDOMNode(this.ref)
+    // close Sidebar if e.target is outside of Sidebar
+    if (!sidebarDOMNode.contains(e.target)) {
+      this.trySetState({ visible: false })
```

Comment:
`visible` is destructured from `props` in the `render()` method.  It should, in fact, be from `state` since it is auto controlled.